packages/frontend/src/features/semanticViewer/components/ContentCharts.tsx

```diff
@@ -113,50 +168,55 @@ const ContentCharts: FC = () => {
-                        />
-
-                        <Panel
-                            id={`semantic-viewer-panel-tab-${TabPanel.VISUALIZATION_TABLE}`}
-                            collapsible
-                            order={2}
-                            defaultSize={25}
-                            minSize={10}
-                            onCollapse={() => setOpenPanel(undefined)}
-                        >
-                            <Table
-                                resultsRunner={pivotResultsRunner}
-                                columnsConfig={Object.fromEntries(
-                                    chartVizQuery.data?.columns.map((field) => [
-                                        field.reference,
-                                        {
-                                            visible: true,
-                                            reference: field.reference,
-                                            label: field.reference,
-                                            frozen: false,
-                                            // TODO: add aggregation
-                                            // aggregation?: VizAggregationOptions;
-                                        },
-                                    ]) ?? [],
-                                )}
+                {openPanel === TabPanel.VISUALIZATION_TABLE &&
+                    !isVizTableConfig(vizConfig) && (
+                        <>
+                            <Box
+                                component={PanelResizeHandle}
+                                bg="gray.2"
+                                h="xs"
+                                sx={(theme) => ({
+                                    transition:
+                                        'background-color 0.2s ease-in-out',
+                                    '&[data-resize-handle-state="hover"]': {
+                                        backgroundColor: theme.colors.gray[3],
+                                    },
+                                    '&[data-resize-handle-state="drag"]': {
+                                        backgroundColor: theme.colors.gray[4],
+                                    },
+                                })}
                             />
-                        </Panel>
-                    </>
-                )}
+
+                            <Panel
+                                id={`semantic-viewer-panel-tab-${TabPanel.VISUALIZATION_TABLE}`}
+                                collapsible
+                                order={2}
+                                defaultSize={25}
+                                minSize={10}
+                                onCollapse={() => setOpenPanel(undefined)}
+                            >
+                                <Table
+                                    resultsRunner={pivotResultsRunner}
+                                    thSortConfig={thSortConfig}
```

Comment:
just throwing some ideas: this does not feel right. I feel like pivot results runner should be responsible to provide both sorting config and columns config. or maybe `TableDataModel` is better suited for that. but we only use TableDataModel for visualization... we might need two more models for raw results and pivot results. happy to discuss.